### PR TITLE
Fix link in disabled events still processed internally

### PR DESCRIPTION
**Pull request by 4ian**
This could generate infinite loop crashs even when a link is included as a sub event of a disabled event.

Add additional tests for safety.

### CODE REVIEW

File: Core/GDCore/Events/Builtin/LinkEvent.cpp [hunk 1]
```diff
@@ -131,30 +131,36 @@ void LinkEvent::SerializeTo(SerializerElement& element) const {
...
-  SerializerElement& includeElement = element.GetChild("include", 0, "Limites");
-
   SetTarget(element.GetChild("target", 0, "Scene").GetValue().GetString());
 
-  if (includeElement.HasAttribute("includeAll")) {
-    // Compatibility with GDevelop <= 4.0.92
-    if (includeElement.GetBoolAttribute("includeAll", true)) {
-      SetIncludeAllEvents();
+  // Compatibility with GD <= 5
+  if (element.HasChild("include", "Limites")) {
```
Comment (4ian) [inline Core/GDCore/Events/Builtin/LinkEvent.cpp:137]: This is to allow links without this long deprecated field in the JSON

File: Core/GDCore/Events/CodeGeneration/EventsCodeGenerator.cpp [hunk 2]
```diff
@@ -963,6 +963,8 @@ void EventsCodeGenerator::DeleteUselessEvents(gd::EventsList& events) {
  */
 void EventsCodeGenerator::PreprocessEventList(gd::EventsList& listEvent) {
   for (std::size_t i = 0; i < listEvent.GetEventsCount(); ++i) {
+    if (listEvent[i].IsDisabled()) continue;
```
Comment (4ian) [inline Core/GDCore/Events/CodeGeneration/EventsCodeGenerator.cpp:966]: This is the fix. `Preprocess` would skip the event, but not the sub events with the lines below!

File: GDevelop.js/__tests__/GDJSBasicCodeGenerationIntegrationTests.js [hunk 3]
```diff
@@ -30,6 +30,57 @@ describe('libGD.js - GDJS Code Generation integration tests', function () {
     })
   );
 
+  it('does not generate anything for a disabled event and its sub events', function () {
```
Comment (4ian) [inline GDevelop.js/__tests__/GDJSBasicCodeGenerationIntegrationTests.js:33]: This was already working

File: GDevelop.js/__tests__/GDJSLayoutCodeGenerationIntegrationTests.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+          },
+        ],
+        events: [],
+      },
+      // The new event is disabled, so not executed. And the link
+      // should not be inserted (so no infinite loop), because it's not
+      // even processed (as it's inside a disabled event).
+      {
+        type: 'BuiltinCommonInstructions::Standard',
+        disabled: true,
```
Comment (4ian) [inline GDevelop.js/__tests__/GDJSLayoutCodeGenerationIntegrationTests.js:71]: Before the fix, even if this even was disabled, we would have an infinite loop leading to a crash.

File: GDevelop.js/__tests__/GDJSLayoutCodeGenerationIntegrationTests.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    const { gdjs, runtimeScene } = makeMinimalGDJSMock();
+    runCompiledEvents(gdjs, runtimeScene);
+
+    expect(runtimeScene.getVariables().has('Counter')).toBe(true);
+    expect(runtimeScene.getVariables().get('Counter').getAsNumber()).toBe(7);
+
+    project.delete();
+  });
+
+  it('does generate code for nested links to external events', function () {
```
Comment (4ian) [inline GDevelop.js/__tests__/GDJSLayoutCodeGenerationIntegrationTests.js:107]: Just for safety, but was already working.

**Review by D8H — APPROVED** [review]
LGTM

**Pull request merged**